src/synth/fluid_voice.h

```diff
@@ -68,7 +68,7 @@ struct _fluid_voice_t
 	fluid_channel_t* channel;
 	fluid_gen_t gen[GEN_LAST];
 	fluid_mod_t mod[FLUID_NUM_MOD];
-	int mod_count;
+	unsigned int mod_count;
```

Comment:
This one is dangerous. If mod_count somewhere in the code could be negative (for instance as a temporary state), it would become a very large number.

Reply:
At least for the current implementation it should never become negative. Given the fact that it will never become greater `FLUID_NUM_MOD` (==64), we could change it's type to char, as we're having one padding byte over in `fluid_voice_t` anyway.

Reply:
Maybe. My point was that this use of unsigned is more dangerous since the type is not immediately visible from where "mod_count" is actually used, since the declaration is in a different file. In the future, this could theoretically cause a bug.

I've been bitten by unnecessary bugs caused by use of unsigned so many times that it's become a habit of mine to point this out in other peoples code, and never use unsigned in my own code.

Reply:
FWIW: I always disliked using anything but int for "normal" scale numbers, but that was just a gut feeling and sometimes I used unsigned anyway. The case made in the google style guide against unsigned ints sounds very convincing to me. Thanks for the pointer!

Reply:
Ok then, lets define it as `signed char mod_count;` and use an `int` when iterating over it. That way the `mod_count` would be promoted to `int` in any loop.

Reply:
Well, the reason because I proposed to change "mod_count" to unsigned type was not because I really wanted to optimize it, but because I got a complain of sign mismatch with FLUID_N_ELEMENTS(), so I got 2 choices: writing "(int)FLUID_N_ELEMENTS(blablabla)" or adjusting mod_count. I tried to suggest the second choice because it looked more correct to me, after I verified that the fix was safe in all parts of the current source code.

Unsigned types are dangerous if you do not use them correctly. That's true. Is this particular change dangerous? In my opinion, it should not, if mod_count becomes negative (or 0xffffffff if read as unsigned), Fluidsynth will hang in both cases. Actually, Google coding style guide just shows you some dangers could happen *if* the coder does not take the right care, that's true. Even the pointers can dangerous if you do pay attention... 

However, instead of changing to "signed char", I really prefer to revert the change and restore "int".
Byte access is another thing that should be used with some care because it can have several speed penalties on modern CPU.
Since this is not an array and there is not the reason for saving space, I would recommend to use the previous code if this change is not acceptable. If it's ok for you too, I can revert everything and I will cast FLUID_N_ELEMENTS() to (int) type.

Reply:
Byte access might not be the best, however it will only happen once. I dont see a speed penalty here. I would rather prefer to decrease `sizeof(fluid_voice_t)` as there may be many instances of voices alive. But sure, that's not your problem right now. Just change it back to int for now. However, casting `FLUID_N_ELEMENTS` to `int` definitely feels wrong to me.